pkg/cmd/cli/cmd/newapp.go

```diff
@@ -106,6 +106,59 @@ func NewCmdNewApplication(fullName string, f *clientcmd.Factory, out io.Writer)
 
 // RunNewApplication contains all the necessary functionality for the OpenShift cli new-app command
 func RunNewApplication(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args []string, config *newcmd.AppConfig) error {
+	if err := parseParameters(f, c, args, config); err != nil {
+		return err
+	}
+
+	result, err := config.RunAll(out)
+	if err != nil {
+		if errs, ok := err.(errors.Aggregate); ok {
+			if len(errs.Errors()) == 1 {
+				err = errs.Errors()[0]
+			}
+		}
+		if err == newcmd.ErrNoInputs {
```

Comment:
sorry, yup makes sense.... for some reason I thought it was inside the previous if